### PR TITLE
Add data loader for HF oasst1

### PR DESCRIPTION
**Pull request by grgau**
Added a PR as requested in [#2828 ](https://github.com/LAION-AI/Open-Assistant/issues/2828)

Changes in this PR:
* Add to `oasst_export` configurations the attributes `hf_dataset_name` and `use_hf_dataset` to declare which HF oasst dataset to use
* Add the loading of `hf_dataset_name` in `load_oasst_export` function, making it possible to work with datasets directly from HuggingFace and not just local datasets.

Very happy to contribute to this project in some way!! expecting your review 😄

### CODE REVIEW

File: model/model_training/custom_datasets/oasst_dataset.py [hunk 1]
```diff
@@ -35,6 +38,10 @@ def load_oasst_export(
     generator = Generator()
     generator.manual_seed(manual_seed)
 
+    if use_hf_dataset and hf_dataset_name:
+        hf_dataset = load_dataset(hf_dataset_name)
+        return hf_dataset["train"], hf_dataset["validation"]
```
Comment (andreaskoepf) [inline model/model_training/custom_datasets/oasst_dataset.py:43]: The format between trees and messages in the HF dataset differs greatly .. some conversion is required to restore the OA format.

Comment (grgau) [reply model/model_training/custom_datasets/oasst_dataset.py:43]: any tips/suggestions about it? first time working on an issue

**Comment by grgau** [comment]
Had an idea, still working on this

File: oasst-data/oasst_data/schemas.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ class LabelAvgValue(BaseModel):
     count: int
 
 
-LabelValues = dict[str, LabelAvgValue]
```
Comment (olliestanley) [inline oasst-data/oasst_data/schemas.py:14]: What is the reason for this change (and the similar one for `emojis` below)?

Comment (grgau) [reply oasst-data/oasst_data/schemas.py:14]: `emojis` and `labels` fields has a slightly different structure in loaded HF dataset, for example

original jsonl dataset file
``` json
        "emojis": {
            "+1": 10,
            "_skip_reply": 1,
            "_skip_ranking": 4
        },
```

HF loaded dataset
```json
        "emojis": {
            "name": [
                "+1",
                "_skip_reply",
                "_skip_ranking"
            ],
            "count": [
                10,
                1,
                4
            ]
        },

```

So I did these changes to be possible to parse with pydantic in both scenarios. I though about doing some transformations on HF structure but it would make the loading process slower.

**Comment by grgau** [comment]
@olliestanley any updates about it?

**Comment by olliestanley** [comment]
> @olliestanley any updates about it?

Will need reviewing by ML code owner(s), I am not one of them. I'll remind them

**Comment by andreaskoepf** [comment]
.. loading needs to restore the ORIGINAL format. Sorry, forgot ab about it.

**Comment by grgau** [comment]
@andreaskoepf I will work on this

**Comment by andreaskoepf** [comment]
I started already working on it myself …

**Review by andreaskoepf — APPROVED** [review]
I think it would be good to go .. would be great if someone else could review the changes quickly..
